### PR TITLE
Delete feature

### PR DESCRIPTION
**Pull request by stdavis**
I think that this is pretty close. Just need to get the exact signature for the delete service.

### CODE REVIEW

File: src/app/project/FeatureDetails.js [hunk 1]
```diff
@@ -146,6 +182,28 @@ define([
...
+        },
+        onDeleteFeatureClick: function () {
+            // summary:
+            //      user has clicked the delete feature button
+            console.log('app.project.FeatureDetails:onDeleteFeatureClick', arguments);
+
+            var that = this;
+            var projectId = router.getProjectId();
+            var onErrorWithDefault = lang.partial(onError, 'Error deleting feature!');
+            xhr.del(config.urls.api + '/project/' + projectId + '/feature/delete/' + that.currentId, {
```
Comment (steveoh) [inline src/app/project/FeatureDetails.js]: I would remove the delete fragment and allow the verb to dictate what is happening. I think it should be the same signature as the feature details with the query string category

Comment (stdavis) [inline src/app/project/FeatureDetails.js]: Addressed.

**Comment by steveoh** [comment]
I think we should add a delete confirmation dialog.

**Comment by stdavis** [comment]
Great point. Added with ab8418c.

**Comment by steveoh** [comment]
We also need the user token and key for authentication/authorization.

**Comment by stdavis** [comment]
Addressed in fb8598f.

**Comment by steveoh** [comment]
we should also disable the button to guard against double clickers.

**Comment by steveoh** [comment]
do we also need to refresh the layer?

**Comment by stdavis** [comment]
> disable the button to guard against double clickers

Addressed in 672f119

> show notification of the inflight request

Addressed in a2d1bc1

> change success http status code to 202

Addressed in 5974897

> possibly refresh the layer after success?

That's what [this line](https://github.com/agrc/wri-web/blob/delete-feature/src/app/project/FeatureDetails.js#L207) should do. Is it not working?

**Pull request merged**